src/ipa/rpi/controller/rpi/sync.cpp

```diff
@@ -119,100 +144,198 @@ void Sync::initialise()
+
 		if (!syncReady_ && !(readyFrame_ - frameCount_)) {
-			LOG(RPiSync, Info) << "Sync ready at frame " << frameCount_ << " ts " <<  payload.wallClock;
-			syncReady_ = true;
+			if (local.wallClock > syncTime - frameDuration_.get<std::micro>() * 0.5) {
+				LOG(RPiSync, Error) << "Sync ready is true";
+				syncReady_ = true;
+				lag = local.wallClock - syncTime;
+				if (local.wallClock > syncTime + frameDuration_.get<std::micro>() * 0.5) {
+					LOG(RPiSync, Warning) << "Frame has been lost, sync started with lag of: " << lag << " us";
+				} else {
+					LOG(RPiSync, Info) << "Sync started without lag";
+				}
+			}
+		} else if (!syncReady_) {
+			syncTime = local.wallClock + frameDuration_.get<std::micro>() * (readyFrame_ - frameCount_);
 		}
 
 		if (!(frameCount_ % syncPeriod_)) {
-			static int lastWallClock = local.wallClock;
+			static int64_t nextSensorTimestamp = local.wallClock;
 			payload.sequence = local.sequence;
-			payload.wallClock = local.wallClock;
+			payload.wallClock = trendingClock.modeled_wall_clock(local.wallClock, local.sensorTimestamp, local.sequence);
+			payload.sensorTimestamp = local.sensorTimestamp;
 			payload.nextSequence = local.sequence + syncPeriod_;
-			payload.nextWallClock = local.wallClock + frameDuration_.get<std::micro>() * syncPeriod_;
+			payload.nextWallClock = payload.wallClock + frameDuration_.get<std::micro>() * syncPeriod_;
 			payload.readyFrame = std::max<int32_t>(0, readyFrame_ - frameCount_);
 
-			int jitter = payload.wallClock - lastWallClock;
-			lastWallClock = payload.nextWallClock;
+			//jitter without sensor timestmp will wall clock, sine nt big eal with sensor timestamp
+			int64_t jitter = payload.wallClock - nextSensorTimestamp;
+			nextSensorTimestamp = payload.nextWallClock;
 
-			if (sendto(socket_, &payload, sizeof(payload), 0, (const sockaddr *)&addr_, sizeof(addr_)) < 0)
-				LOG(RPiSync, Error) << "Send error!";
-			else
-				LOG(RPiSync, Info) << "Sent message: seq " << payload.sequence << " ts " << payload.wallClock << " jitter " << jitter << "us"
-						<< " : next seq " << payload.nextSequence << " ts " << payload.nextWallClock << " : ready frame " << payload.readyFrame;
+			if (sendto(socket_, &payload, sizeof(payload), 0, (const sockaddr *)&addr_, sizeof(addr_)) < 0) {
+				LOG(RPiSync, Error) << "Send error! " << strerror(errno);
+			} else {
+				LOG(RPiSync, Info) << "Sent message: seq " << payload.sequence << " jitter " << jitter << "us" 
+									<< " : ready frame " << payload.readyFrame;
+			}
 		}
 	} else if (mode_ == Mode::Client) {
 
+		static ClockRecovery trendig_error(syncPeriod_, lineFitting_);
+		static ClockRecovery trendingClock(local.wallClock, local.sensorTimestamp, syncPeriod_, lineFitting_);
 		static int frames = 0;
 		socklen_t addrlen = sizeof(addr_);
+		static bool IPCheck = false;
+		static std::string usingWallClock_ = "yes";
+
+		/* Approximates frame duration */
+		static int64_t modeled_wall_clock_value = 0;
```

Comment:
Coding style generally wants camel case. So this would be `modeledWallClockValue`.  Unfortunately this applies pretty much everywhere...

I think it's probably "modelled" rather than "modeled" too?